src/server/frontend_wayland/generator/enum.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#include "enum.h"
+#include "utils.h"
+
+#include <libxml++/libxml++.h>
+
+Enum::Enum(xmlpp::Element const& node)
+    : name{sanitize_name(to_camel_case(node.get_attribute_value("name")))}
+{
+    for (auto const& child : node.get_children("entry"))
+    {
+        auto entry_node = dynamic_cast<xmlpp::Element const*>(child);
+        std::string entry_name = to_upper_case(entry_node->get_attribute_value("name"));
```

Comment:
This doesn't match the Mir style guide. These elements should be `lower_case`.

Reply:
Fixed